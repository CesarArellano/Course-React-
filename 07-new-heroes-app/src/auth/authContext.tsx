import { createContext } from "react";

export interface AuthState {
  user: {
    name: string;
    logged: boolean;
  }
  dispatch: React.Dispatch<any>
}

export const AuthContext = createContext({} as AuthState );