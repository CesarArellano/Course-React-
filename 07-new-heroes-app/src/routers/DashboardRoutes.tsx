import { Route, Routes } from 'react-router-dom';

import { Navbar } from '../components/Navbar';
import { DCScreen } from '../components/dc/DCScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <Routes>
        <Route path="/marvel" element={ <MarvelScreen /> } />
        <Route path="/dc" element={ <DCScreen /> } />
        <Route path="/search" element={ <SearchScreen /> } />
        <Route path="/hero" element={ <HeroScreen /> } />
        <Route path="/" element={ <MarvelScreen /> } />
      </Routes>
    </>
  )
}
