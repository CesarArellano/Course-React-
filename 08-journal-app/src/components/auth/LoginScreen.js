import React from 'react'
import { Link } from 'react-router-dom'

export const LoginScreen = () => {
  return (
    <>
      <div className="center-box">
        <img 
          src="https://image.flaticon.com/icons/png/512/3235/3235042.png"
          alt="Journal App Icon" 
          title="App Icon"
          className="img-icon"
        />
      </div>
      <h3 className="auth__title">Login</h3>
      <form>
        <input
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          className="auth__input"
        />

        <input 
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
        />

        <div className="center-box">
          <button
            type="submit"
            className="btn btn-primary mb-5"
          >
            Login
          </button>
        </div>
        <hr />
        <div className="auth__social-networks">
          <p>Login with social networks</p>
          <div className="google-btn">
              <div className="google-icon-wrapper">
                  <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
              </div>
              <p className="btn-text">
                  <b>Sign in with google</b>
              </p>
          </div>
        </div>
          <Link 
            to="/auth/register"
            className="link"
          >
            Create New Account
          </Link>
      </form>
    </>
  )
}
