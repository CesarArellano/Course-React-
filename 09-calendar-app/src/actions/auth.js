import { fetchWithoutToken } from "../helpers/fetch";
import { types } from "../types/types";

export const startLogin = ( email, password ) => {
  return async ( dispatch ) => {
    const resp = await fetchWithoutToken('auth', { email, password }, 'POST');
    const decodedData = await resp.json();
    
    if( decodedData.ok ) {
      localStorage.setItem( 'token', decodedData.token );
      localStorage.setItem( 'token-init-date', new Date().getTime() );
      dispatch( login({
        uid: decodedData.uid,
        name: decodedData.name,
      }));
    }
  }
};

const login = ( user ) => ({
  type: types.authLogin,
  payload: user,
});