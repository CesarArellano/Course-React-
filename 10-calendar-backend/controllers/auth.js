const { response } = require('express');
const User = require('../models/User');

const createUser = async ( req, res = response ) => {
  
  const { email, password } = req.body;
  try {
    let user = await User.findOne({ email })

    if( user ) {
      return res.status(400).json({
        ok: false,
        msg: 'El usuario ya existe con ese correo.'
      });
    }

    user = new User( req.body );
    user.save();
    
    res.status(201).json({
      ok: true,
      uid: user.id,
      name: user.name
    });

  } catch(error) {
    console.log(error);
    res.status(500).json({
      ok: false,
      msg: 'Please talk to the admin'
    });
  }
  
}

const userLogin = ( req, res = response ) => {
  const { email, password } = req.body;

  res.json({
    ok: true,
    msg: 'login',
    email,
    password
  });

};

const revalidateToken = ( req, res = response ) => {
  res.json({
    ok: true,
    msg: 'renew'
  });
};

module.exports = {
  createUser,
  userLogin,
  revalidateToken
};