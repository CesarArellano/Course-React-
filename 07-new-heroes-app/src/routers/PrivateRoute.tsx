import React, { useContext } from 'react'
import { Navigate } from 'react-router-dom';
import { AuthContext } from '../auth/authContext'

export const PrivateRoute = ({ children }: any) => {
  const { user } = useContext( AuthContext );
  
  return user.logged
    ? <> { children } </>
    : <Navigate to="/login "/>
}
