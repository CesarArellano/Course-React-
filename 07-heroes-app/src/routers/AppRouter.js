import React, { useContext } from 'react'
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
  const { user:{ logged } } = useContext(AuthContext);
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute 
            exact 
            path="/login" 
            isAuthenticated={ logged }
            component={ LoginScreen } 
          />
          <PrivateRoute 
            path="/" 
            isAuthenticated={ logged }
            component={ DashboardRoutes } 
          />
        </Switch>
      </div>
    </Router>
  );
}
