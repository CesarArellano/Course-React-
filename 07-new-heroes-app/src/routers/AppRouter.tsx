import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { LoginScreen } from '../components/login/LoginScreen';
import { PublicRoute } from './PublicRoute';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/login" element={ 
          <PublicRoute>
            <LoginScreen />
          </PublicRoute>
          } 
        />
        <Route path="/*" element={ 
            <PrivateRoute>
              <DashboardRoutes />
            </PrivateRoute>
          }
        />
        {/* <Route path="/*" element={ <DashboardRoutes /> }/> */}
      </Routes>
    </BrowserRouter>
  );
}
