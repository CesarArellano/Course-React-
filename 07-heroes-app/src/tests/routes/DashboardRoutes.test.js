import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

describe('Testing <DashboardRoutes />', () => {
  test('should show it correctly', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: 'César',
        logged: false
      }
    };

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>  
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('César')
  });
  
});
