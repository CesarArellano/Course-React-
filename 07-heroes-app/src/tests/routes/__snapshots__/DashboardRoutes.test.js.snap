// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <DashboardRoutes /> should show it correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
      </div>
      <div
        className="d-flex"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <span
            className="nav-item nav-link text-info"
          >
            CÃ©sar
          </span>
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Spider Man"
                className="card-img-top"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Captain America"
                className="card-img-top"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Iron Man"
                className="card-img-top"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Thor"
                className="card-img-top"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Hulk"
                className="card-img-top"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Wolverine"
                className="card-img-top"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Daredevil"
                className="card-img-top"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Hawkeye"
                className="card-img-top"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Cyclops"
                className="card-img-top"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxwidth": 540,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Imagen Silver Surfer"
                className="card-img-top"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
